org.sil.scripture-app-builder.yml

```diff
@@ -0,0 +1,114 @@
+# REFERENCES:
+  # Python2.7:  https://github.com/flathub/shared-modules/blob/cc50bdbb34cc3d29bf61cfea5e27cd980b7bda04/python2.7/python-2.7.json
+  # OpenJDK17: https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk17/blob/branch/23.08/org.freedesktop.Sdk.Extension.openjdk17.yaml
+  # Node20:    https://github.com/flathub/org.freedesktop.Sdk.Extension.node20/blob/branch/23.08/org.freedesktop.Sdk.Extension.node20.yaml
+# FIRST-TIME-BUILD: ~30m on 10-20Mb/s download speed
+app-id: org.sil.scripture-app-builder
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+  # - org.freedesktop.Sdk.Extension.openjdk21
+finish-args:
+  # Window x11 socket access
+  - --socket=x11
+  # Audio access
+  - --socket=pulseaudio
+  # Unsure if the program absolutely needs ssh
+  # - --socket=ssh-auth
+  # Or the printer socket
+  # - --socket=cups
+  # Network access
+  - --share=network
+  # XShm access
+  - --share=ipc
+  # All external devices (catch all for devices we interact with and graphical acceleration)
+  - --device=all
+  # Local file browsing access
+  - --filesystem=host
+  # Temporary file access
+  - --filesystem=/tmp
```

Comment:
why is /tmp needed?

Reply:
The build process writes files to /tmp.

Reply:
These permissions have no effect on the build process.

Reply:
Sorry, when I said the build process, I should have said the mobile app build process which is done during the running of the desktop app.

Reply:
> Sorry, when I said the build process, I should have said the mobile app build process which is done during the running of the desktop app.

Does it need to be `/tmp` from host? You can set the TMPDIR variable to XDG_RUNTIME_DIR or `/var/tmp`. One is tmpfs, the second one is persistent.

Reply:
I did some testing.  Access to /tmp was being used for builds.  I changed the sab-run.sh to have `-Djava.io.tmpdir=/var/tmp` so that the default tmp dir is set to /var/tmp instead of /tmp.